addons/account/models/account_move.py

```diff
@@ -317,8 +318,14 @@ def action_post(self):
 
     @api.multi
     def button_cancel(self):
+        AccountMoveLine = self.env['account.move.line']
+        excluded_move_ids = []
+
+        if self._context.get('edition_mode'):
+            excluded_move_ids = AccountMoveLine.search(AccountMoveLine._get_domain_for_edition_mode()).mapped('move_id').ids
+
         for move in self:
-            if not move.journal_id.update_posted:
+            if not move.journal_id.update_posted and move.id not in excluded_move_ids:
```

Comment:
Check the condition here move.to_check == True, then the lines to avoid loading the ids.

Reply:
I changed to 
```diff
        if self._context.get('edition_mode'):
-            excluded_move_ids = AccountMoveLine.search(AccountMoveLine._get_domain_for_edition_mode()).mapped('move_id').ids
+            excluded_move_ids = AccountMoveLine.search(AccountMoveLine._get_domain_for_edition_mode() + [('id', 'in', self.ids)]).mapped('move_id').ids
        for move in self:
            if not move.journal_id.update_posted and move.id not in excluded_move_ids:
                ...
```